src/common/cups_print.c

```diff
@@ -167,6 +172,7 @@ static int _dest_cb(void *user_data, unsigned flags, cups_dest_t *dest)
 
 static int _cancel = 0;
 
+// FIXME: if we can guarantee that we have CUPS >= 1.6 and MacOS >= 10.8 this could be much simpler
```

Comment:
CUPS 1.6 was released on the 11 juil. 2013, we can assume that we have at least such version and we require that in the CMake check.

macOS was released on the 25 juillet 2012, so again we certain can expect to have a newer version.

Reply:
Absolutely. I started in on [another PR](https://github.com/dtorop/darktable/compare/print-view-discovery-fix...dtorop:darktable:print-cups-versioning?expand=1) to fix this up and remove a bunch of obsolete ifdef's. The easy part is to require CUPS >= 1.7 and no longer handle macOS <= 10.8.

I still need to follow up on your note (from 2018) at the start of cups_print.c:

``` c
// some platforms are starting to provide CUPS 2.2.9 and there the
// CUPS API deprecated routines ate now flagged as such and reported as
// warning preventing the compilation.
//
// this seems wrong and PPD should be removed from this unit. but there
// still one missing piece discussed with the CUPS maintainers about the
// way to get media-type using the IPP API. nothing close to working at
// this stage, so instead of breaking the compilation on platforms using
// recent CUPS version we kill the warning.
```

So far I've gotten that `cupsGetPPD()` is deprecated and the advice is to use `cupsCopyDestInfo()`. Do you remember details? Regardless, I can get to this, I just need to test to see if the non-deprecated CUPS API now works.

I think this is work for another PR. Are you OK with the FIXME notice in there as a placeholdeer for work to be done? I don't think this is urgent (it's not a bugfix unlike this PR), but it would be an improvement.